rpki-validator/src/main/java/net/ripe/rpki/validator3/api/roas/ExportsController.java

```diff
@@ -133,6 +142,32 @@ public void exportCsv(HttpServletResponse response) throws IOException {
         }
     }
 
+    @ApiOperation("export VRPs (json) extended with validity and serial number")
+    @GetMapping(path = "/api/export-extended.json")
+    public JsonExportExtended exportJsonExtended(HttpServletResponse response) {
+        response.setContentType(JSON);
+
+        if (!storage.readTx(settings::isInitialValidationRunCompleted)) {
+            response.setStatus(HttpServletResponse.SC_NO_CONTENT);
+            return null;
+        }
+
+        Stream<JsonRoaPrefixExtended> validatedPrefixes = validatedRpkiObjects
+                .findCurrentlyValidatedRoaPrefixes()
+                .getObjects()
+                .map(r -> new JsonRoaPrefixExtended(
+                        String.valueOf(r.getAsn()),
+                        r.getPrefix().toString(),
+                        r.getEffectiveLength(),
+                        r.getTrustAnchor().getName(),
+                        Instant.ofEpochMilli(r.getNotBefore()) + " - " + Instant.ofEpochMilli(r.getNotAfter()),
```

Comment:
I think that as a user I'd rather consume this in two separate fields.

Reply:
Separated them.